src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -266,7 +266,7 @@ public uint WritePacket(SNIHandle handle, SNIPacket packet, bool sync)
         private static string GetServerNameWithOutProtocol(string fullServerName, string protocolHeader)
         {
             string serverNameWithOutProtocol = null;
-            if (fullServerName.Length > protocolHeader.Length &&
+            if (fullServerName.Length >= protocolHeader.Length &&
```

Comment:
There's this test case in the named pipes test:
DataSource = "np:"

(invalid connection string error in this case)